index/scorch/segment/zap/contentcoder.go

```diff
@@ -146,33 +161,55 @@ func (c *chunkedContentCoder) Add(docNum uint64, vals []byte) error {
 }
 
 // Write commits all the encoded chunked contents to the provided writer.
-func (c *chunkedContentCoder) Write(w io.Writer) (int, error) {
+//
+// | ..... data ..... | chunk offsets (varints)
+// | position of chunk offsets (uint64) | number of offsets (uint64) |
+//
+func (c *chunkedContentCoder) Write() (int, error) {
 	var tw int
-	buf := make([]byte, binary.MaxVarintLen64)
-	// write out the number of chunks
-	n := binary.PutUvarint(buf, uint64(len(c.chunkLens)))
-	nw, err := w.Write(buf[:n])
-	tw += nw
-	if err != nil {
-		return tw, err
+
+	if c.final != nil {
+		// write out the data section first
+		nw, err := c.w.Write(c.final)
+		tw += nw
+		if err != nil {
+			return tw, err
+		}
 	}
 
+	chunkOffsetsStart := uint64(tw)
+
+	buf := make([]byte, binary.MaxVarintLen64)
 	chunkOffsets := modifyLengthsToEndOffsets(c.chunkLens)
 	// write out the chunk offsets
 	for _, chunkOffset := range chunkOffsets {
 		n := binary.PutUvarint(buf, chunkOffset)
-		nw, err = w.Write(buf[:n])
+		nw, err := c.w.Write(buf[:n])
 		tw += nw
 		if err != nil {
 			return tw, err
 		}
 	}
-	// write out the data
-	nw, err = w.Write(c.final)
+
+	chunkOffsetsLen := uint64(tw) - chunkOffsetsStart
+
+	buf = make([]byte, 8)
```

Comment:
should be able to reuse the buf allocated from line 182, resliced to 8 bytes length

Reply:
Done.